app/services/search/query_builders/chat_channel_membership.rb

```diff
@@ -0,0 +1,77 @@
+module Search
+  module QueryBuilders
+    class ChatChannelMembership
+      FILTER_KEYS = %i[
+        channel_status
+        channel_type
+        status
+        viewable_by
+      ].freeze
+
+      QUERY_KEYS = %i[
```

Comment:
These are fields we want to run text searches on, ie score documents based on how well they match. For more information on queries vs filters [checkout the Elasticsearch docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-filter-context.html)